node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -394,6 +398,11 @@ object Main {
     } yield kademliaPortConf
   }
 
+  private def checkShardNameOnlyAscii[F[_]: Sync: Log](shardName: String): F[Unit] =
+    (Log[F].error("Shard name should contain only ASCII characters") >>
+      Sync[F].raiseError(new RuntimeException("Invalid shard name")))
+      .whenA(!CharMatcher.ascii().matchesAllOf(shardName))
```

Comment:
We need this check on all places where shard ID can be supplied from external source or used for some specific purpose.

So precisely checking must be done in:
- when node starts (it's required data, node cannot work without shard ID)
- block validation when shard name is checked
- in creation of block random seed.

Reply:
@tgrospic, I've updated PR. Should we additionally check shard name during making deploy?